presto-pinot-toolkit/src/main/java/com/facebook/presto/pinot/query/PinotQueryGenerator.java

```diff
@@ -146,32 +145,52 @@ public PinotQueryGeneratorContext getContext()
         }
     }
 
-    public static class GeneratedPql
+    public enum PinotQueryFormat {
+        PQL,
+        SQL
+    }
+
+    public static class GeneratedPinotQuery
     {
         final String table;
+        final String sql;
         final String pql;
         final List<Integer> expectedColumnIndices;
         final int groupByClauses;
         final boolean haveFilter;
         final boolean isQueryShort;
 
         @JsonCreator
-        public GeneratedPql(
+        public GeneratedPinotQuery(
                 @JsonProperty("table") String table,
+                @JsonProperty("sql") String sql,
                 @JsonProperty("pql") String pql,
                 @JsonProperty("expectedColumnIndices") List<Integer> expectedColumnIndices,
                 @JsonProperty("groupByClauses") int groupByClauses,
                 @JsonProperty("haveFilter") boolean haveFilter,
                 @JsonProperty("isQueryShort") boolean isQueryShort)
         {
             this.table = table;
+            this.sql = sql;
             this.pql = pql;
             this.expectedColumnIndices = expectedColumnIndices;
```

Comment:
Should we check state that exactly one of 'sql' or 'pql' is set ?

Reply:
will do

Reply:
added the check for exact one query to be set.